File: DecompiledGameScripts/cyberpunk/devices/blindingLights/reflector.swift

public class Reflector extends BlindingLight {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as ReflectorController;
  }

  public const func GetDevicePS() -> ref<ReflectorControllerPS> {
    return this.GetController().GetPS();
  }

  private const func GetController() -> ref<ReflectorController> {
    return this.m_controller as ReflectorController;
  }

  protected cb func OnDistraction(evt: ref<Distraction>) -> Bool {
    if evt.IsStarted() {
      this.StartDistraction(true);
    } else {
      this.StopDistraction();
    };
    this.RefreshInteraction(gamedeviceRequestType.Direct, GetPlayer(this.GetGame()));
  }
}

File: Definitions/Game_Definitions/classes/Reflector.lua
---@meta
---@diagnostic disable

---@class Reflector : BlindingLight
Reflector = {}

---@return Reflector
function Reflector.new() return end

---@param props table
---@return Reflector
function Reflector.new(props) return end

---@param evt Distraction
---@return Bool
function Reflector:OnDistraction(evt) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function Reflector:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function Reflector:OnTakeControl(ri) return end

---@return ReflectorController
function Reflector:GetController() return end

---@return ReflectorControllerPS
function Reflector:GetDevicePS() return end

